ci/jenkins/README.md

```diff
@@ -1 +1,96 @@
+  - [Ansible](#ansible)
+  - [Jenkins Job](#jenkins-job)
+    - [Requirements](#requirements)
+    - [Apply the jobs](#apply-the-jobs)
+    - [Jenkins job updater](#jenkins-job-updater)
+- [List of jenkins job](#list-of-jenkins-jobs)
+<!-- /toc -->
+
+## Introduction
+
+This directory includes all the scripts required to run CI for nephe.
+
+We have a cluster of static VMs configured to run as jenkins node.  When a jenkins
+job is triggered, the nephe code will be cloned on the static VM. The jenkins node
+will call terraform to a deploy dynamic VM and use ansible for installing any missing
+packages like docker. Then tests will be run on that dynamic VM, Upon completion
+of the test (success or fail), the dynamic VM will be destroyed.
+
+## Jenkins Deployment
+
+### Teraform
+
+[nephi-ci.sh](./nephe-ci.sh) is responsible for creating terraform tfvars files.
+[deploy.sh](./deploy.sh)is the script to create vm, and store the terraform state
+in a specified directory, which can be used to get VM information later.
+[destroy.sh](./destroy.sh) helps to detroy the VM and delete terraform workspace as well.
+
+For more details about terraform, please refer [here](https://www.terraform.io/docs).
+
+### Ansible
+
+Ansible is used to configure VM once terraform deploys the dynamic VM. All anisble
+tasks are in playbook/roles. For example, we add task to install docker in
+[playbook/roles/docker/tasks/main.yml](playbook/roles/docker/tasks/main.yml).
+
+Ansible is used to copy ssh keys to the dynamic VM, which is used to run the tests
+and collect logs from the dynamic VM later.
+
+For more details about ansible, please refer [here](https://docs.ansible.com/).
+
+### Jenkins Job
+
+We use jenkins job-builder to manage jenkins jobs, the codes are in the [directory jobs](./jobs).
+
+#### Requirements
+
+Yaml files under current directory can be generated via jenkins-job-builder.
+If you want to try out the tests on your local jenkins setup, please check
+the following requirements:
+
+* Jenkins setup
+  * Plugins: ghprb, throttle-concurrents
+  * Jenkins IP/User/Token: they are store in jenkins, and exported to test scripts.
+* Install
+  [jenkins-job-builder](https://jenkins-job-builder.readthedocs.io/en/latest/index.html)
+* Define your `ANTREA_GIT_CREDENTIAL` which is the credential for your private
+  repo.
+* Define your `ghpr_auth`, `nephe_admin_list`, `nephe_org_list` and
+  `nephe_white_list` as
+  [defaults](https://jenkins-job-builder.readthedocs.io/en/latest/definition.html#defaults-1s)
+  variables in a separate file.
+
+#### Apply the jobs
+
+Run the command to test if jobs can be generated correctly.
+
+```bash
+jenkins-jobs test -r ci/jenkins/jobs
+```
+
+Run the command to apply these jobs.
+
+```bash
+jenkins-jobs update -r ci/jenkins/jobs
+```
+
+#### Jenkins job updater
+
+To follow GitOps best practices, there is a job-updater job in Jenkins to detect
+any change in [jobs](./jobs) for every 15 min. As long as a PR to modify code under
```

Comment:
When a PR containing code changes to jobs[link] dir is merged, Jenkins jobs on cloud will be automatically updated with the new changes.

Reply:
done